account_invoice_production_lot/models/account_invoice.py

```diff
@@ -33,19 +34,42 @@ class AccountInvoiceLine(models.Model):
     @api.multi
     def _compute_prod_lots(self):
         for line in self:
-            if not line.order_line_ids:
-                return
-            line.prod_lot_ids = self.mapped(
-                'order_line_ids.procurement_ids.move_ids.lot_ids')
+            line.prod_lot_ids = line.move_line_ids.mapped('lot_ids')
```

Comment:
This could actually be part of the dependency you added `stock_picking_invoice_link`, but let's not take this too far!